ipset.go

```diff
@@ -268,59 +137,224 @@ func (s *IPSet) Ranges() []IPRange {
 	}
 	if len(in) > 0 {
 		// Ran out of removals before the end of in.
-		ret = append(ret, in...)
+		min = append(min, in...)
 		if debug {
-			debugf("ret=%v", ret)
+			debugf("min=%v", min)
 		}
 	}
 
-	// TODO: possibly update s.in and s.out, if #110 supports that.
+	s.in = min
+	s.out = nil
+}
+
+// Add adds ip to the set s.
+func (s *IPSetBuilder) Add(ip IP) { s.AddRange(IPRange{ip, ip}) }
+
+// AddPrefix adds p's range to s.
+func (s *IPSetBuilder) AddPrefix(p IPPrefix) { s.AddRange(p.Range()) }
 
-	return ret
+// AddRange adds r to s.
```

Comment:
If r is not Valid, AddRange does nothing.

?

Reply:
Reminds me I want to maybe change that, the whole "mysteriously do nothing on invalid input" makes me a bit nervous. Filing bug about that.